File: Lab6/src/test/java/com/example/lab6/data/DataBaseRepositoryTest.java
package com.example.lab6.data;

import org.junit.jupiter.api.Test;

import java.sql.SQLException;

import static org.junit.jupiter.api.Assertions.*;

class DataBaseRepositoryTest {

    @Test
    void testDriver() {
        DataBaseConnector testConnector = new DataBaseConnector(("Apartments"));
        assertTrue((testConnector.testDriver()));
    }

    @Test
    void getConnection() {
        DataBaseConnector testConnector = new DataBaseConnector(("Apartments"));
        try {
            assertNotNull((testConnector.getConnection()));
        } catch (SQLException e) {
            fail();
        }
    }
    @Test
    void addApartment() {
        DataBaseRepository repository = new DataBaseRepository(
                new DataBaseConnector("Apartments"));
        repository.addApartment(new Apartment(7, "address7", 1, 10000, false, "", ""));
        repository.addApartment(new Apartment(8, "address8", 2, 20000, true, "date rent8", "term rent38"));
        repository.addApartment(new Apartment(9, "address9", 3, 30000, false, "", ""));
        repository.addApartment(new Apartment(10, "address10", 4, 40000, true, "date rent10", "term rent10"));
        repository.addApartment(new Apartment(11, "address11", 5, 50000, false, "", ""));
        repository.addApartment(new Apartment(12, "address12", 6, 60000, true, "date rent12", "term rent12"));

        assertNotNull(repository.getById(2));
        java.util.List<Apartment> apartments = repository.getAll();
        assertTrue(apartments.size()>0);
    }

    @Test
    void getAll() {
        DataBaseRepository repository = new DataBaseRepository(
                new DataBaseConnector("Apartments"));
        System.out.println(repository.getAll());
    }

    @Test
    void getById() {
        DataBaseRepository repository = new DataBaseRepository(
                new DataBaseConnector("Apartments"));
        System.out.println(repository.getById(1));
    }

    @Test
    void getAllAvailableTrue() {
        DataBaseRepository repository = new DataBaseRepository(
                new DataBaseConnector("Apartments"));
        System.out.println("true " + repository.getAllAvailable(true));
    }
    @Test
    void getAllAvailableFalse() {
        DataBaseRepository repository = new DataBaseRepository(
                new DataBaseConnector("Apartments"));
        System.out.println("false " + repository.getAllAvailable(false));
    }


   /* @Test
    void deleteApartment() {
        DataBaseRepository repository = new DataBaseRepository(
                new DataBaseConnector("test2"));
        repository.deleteApartment(1);
        assertNull(repository.getById(1));
    }*/
}

File: Lab6kotlin/src/test/java/com/example/lab6/data/DataBaseRepositoryTest.kt
package com.example.lab6.data

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import java.sql.SQLException

class DataBaseRepositoryTest {

    @Test
    fun testDriver() {
        val testConnector = DataBaseConnector("Apartments")
        assertTrue(testConnector.testDriver())
    }

    @Test
    fun getConnection() {
        val testConnector = DataBaseConnector("Apartments")
        try {
            assertNotNull(testConnector.connection)
        } catch (e: SQLException) {
            fail<Any>()
        }
    }

    @Test
    fun getAll() {
        val repository = DataBaseRepository(
                DataBaseConnector("Apartments"))
        println(repository.getAll())
    }

    @Test
    fun getById() {
        val repository = DataBaseRepository(
                DataBaseConnector("Apartments"))
        println(repository.getById(2));
    }

    @Test
    fun getAllAvailableTrue() {
        val repository = DataBaseRepository(
                DataBaseConnector("Apartments"))
        println("true " + repository.getAllAvailable(true))
    }

    @Test
    fun getAllAvailableFalse() {
        val repository = DataBaseRepository(
                DataBaseConnector("Apartments"))
        println("false " + repository.getAllAvailable(false))
    }

  /*  @Test
    fun addApartment() {
        val repository = DataBaseRepository(
                DataBaseConnector("Apartments"))
        repository.addApartment(Apartment( "address5", 3, 30000, false, "", ""))
        assertNotNull(repository.getById(2))
        val apartments = repository.getAll()
        assertTrue(apartments.size > 0)
    }

   @Test
    fun deleteApartment() {
       val repository = DataBaseRepository(
               DataBaseConnector("Apartments"))
       repository.deleteApartment(4)
       assertNull(repository.getById(4))
    }*/
}